pkg/controller/statefulset/stateful_set_utils.go

```diff
@@ -88,6 +91,18 @@ func getPodName(set *apps.StatefulSet, ordinal int) string {
 	return fmt.Sprintf("%s-%d", set.Name, ordinal)
 }
 
+// getClaimPodName gets the name of the Pod associated with the Claim, or an empty string if this doesn't look matching.
+func getClaimPodName(set *apps.StatefulSet, claim *v1.PersistentVolumeClaim) string {
+	podName := ""
+
+	statefulClaimRegex := regexp.MustCompile(fmt.Sprintf(".*-(%s-[0-9]+)$", set.Name))
```

Comment:
Yes, but that's okay, because we inject the actual set name into the regex. Really we're just picking out the ordinal (that's also why we can't precompile the RE).

eg if the claim is foo-bar-baz-8, and set.Name = baz, then the RE is foo-bar-(baz-\d+), and we do the right thing. If set.Name = bar-baz, then the RE is foo-(bar-baz-\d+) and again we're okay.

At least I think so, maybe I missed something.